client.go

```diff
@@ -294,6 +295,27 @@ func (c *Client) SetAuthToken(token string) *Client {
 	return c
 }
 
+// SetAuthScheme method sets the auth token scheme type in the HTTP request. For Example:
+//      Authorization: <auth-scheme-value-set-here> <auth-token-value>
+//
+// For Example: To set the scheme to use OAuth
+//
+// 		client.SetAuthScheme("OAuth")
+//
+// This auth token gets added to all the request rasied from this client instance.
+// Also it can be overridden or set one at the request level is supported.
+//
+// Information about Auth schemes can be found in RFC7235 which is linked to below along with the page containing
```

Comment:
You capitalize "Auth" here, don't in most other places, and spell it out in the next line.  Can we be more consistent?  I would personally just spell out the words each time.  It's the authentication scheme on the Authorization header.

Reply:
Done in my branch will push it to rigrassm:master